File: src/SetMismatch.java
public class SetMismatch {
    public int[] findErrorNums(int[] array) {
        int repeated = 0, missing = 0;
        for (int number : array) {
            if (array[Math.abs(number) - 1] < 0) repeated = number;
            else array[Math.abs(number) - 1] *= -1;
        }
        for (int index = 0 ; index < array.length ; index++) {
            if (array[index] > 0) {
                missing = index + 1;
                break;
            }
        }
        return new int[] {Math.abs(repeated), missing};
    }
}

File: python/set_mismatch.py
from typing import List


class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        repeated, missing = 0, 0
        for number in nums:
            if nums[abs(number) - 1] < 0: repeated = number
            else: nums[abs(number) - 1] *= -1
        for index in range(len(nums)):
            if nums[index] > 0:
                missing = index + 1
                break
        return [abs(repeated), missing]
